File: FoosballApi/Models/MatchModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace FoosballApi.Models
{
    public class MatchModel
    {
        public PlayerModel OffencePlayer1 { get; set; }
        public PlayerModel OffencePlayer2 { get; set; }
        public PlayerModel DefencePlayer1 { get; set; }
        public PlayerModel DefencePlayer2 { get; set; }
        public int ScoreTeam1 { get; set; }
        public int ScoreTeam2 { get; set; }
    }
}

File: FoosballApi/Models/TypeScript/Match.model.ts
import { Player } from 'src/app/shared/models/Player.model'

export class Match {
    
    public id: number = 0;
    public offencePlayer1: Player = null;
    public offencePlayer2: Player = null;
    public defencePlayer1: Player = null;
    public defencePlayer2: Player = null;
    public scoreTeam1: number = 0;
    public scoreTeam2: number = 0;

}